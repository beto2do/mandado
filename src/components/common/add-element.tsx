import { ChangeEventHandler, KeyboardEventHandler, MouseEventHandler } from 'react';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';

export function AddElement({
    text, 
    isEmpty, 
    onChange, 
    pressEnter, 
    addElement
    }: {
    text: string, 
    isEmpty: boolean, 
    onChange: ChangeEventHandler<HTMLInputElement>, 
    pressEnter: KeyboardEventHandler<HTMLDivElement>, 
    addElement: MouseEventHandler<HTMLButtonElement>
    }) {

    return (
    <div className="flex gap-1">
        <TextField
            error={isEmpty}
            label="Add Element"
            value={text}
            helperText={isEmpty ? 'Value is required' : '' }
            onChange={onChange}
            onKeyDown={pressEnter}
        />
        <Button
            className='pointer-events-auto bg-red-500 text-white hover:bg-red-400 hover:border-0 border-0 h-14'
            variant="contained"
            sx={{backgroundColor:'#ef4444'}}
            onClick={addElement}
        >Add</Button>
    </div>
    );
}
