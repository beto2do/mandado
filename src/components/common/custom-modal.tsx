import { ReactElement, JSXElementConstructor } from 'react'
import Modal from '@mui/material/Modal';
import Box from '@mui/material/Box';

const style = {
    position: 'absolute' as 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
};

type CustomModalProps = {
    open: boolean, 
    onClose: ((event: {}, reason: "backdropClick" | "escapeKeyDown") => void) | undefined, 
    children: ReactElement<any, string | JSXElementConstructor<any>> 
}

export function CustomModal({open, onClose, children}: CustomModalProps) {
    return (
        <Modal
        open={open}
        onClose={onClose}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
        >
            <Box 
            className="overflow-hidden rounded-lg bg-white text-left shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg"
            sx={{ ...style, width: 500 }}>
                {children}
            </Box>
        </Modal>
    )
}