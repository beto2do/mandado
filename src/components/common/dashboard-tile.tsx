import { Tile } from "@/models";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Typography from "@mui/material/Typography";
import Link from "next/link";
import { CardActionArea, CardActions } from "@mui/material";
import { getImage } from "@/services";
import Image from "next/image";
import Button from "@mui/material/Button";

export async function DashboardTile({ tile }: { tile: Tile }) {
  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardActionArea>
        <Image src={await getImage(tile.imgUrl)} alt="" width={345} />
        <CardContent>
          <Typography gutterBottom variant="h5" component="div">
            {tile.title}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            {tile.content}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Link href={tile.actionUrl}>
          <Button color="primary">{tile.action}</Button>
        </Link>
      </CardActions>
    </Card>
  );
}
