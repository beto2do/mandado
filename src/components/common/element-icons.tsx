import { MouseEventHandler } from "react";
import { Product, ProductStatus } from "@/models";
import IconButton from "@mui/material/IconButton";
import DeleteIcon from "@mui/icons-material/Delete";
import EditIcon from "@mui/icons-material/Edit";
import AddCirculeIcon from "@mui/icons-material/AddCircle";

type ElementIconsProps = {
  product: Product;
  onEditEvent?: MouseEventHandler<Element> | undefined;
  onDeleteEvent?: MouseEventHandler<Element> | undefined;
  onSaveEvent?: MouseEventHandler<Element> | undefined;
};

export function ElementIcons({
  product,
  onEditEvent,
  onDeleteEvent,
  onSaveEvent,
}: ElementIconsProps) {
  const isEditionStatus = product.status === ProductStatus.EDITION;
  const editionView = isEditionStatus ? (
    <IconButton edge="end" aria-label="comments" onClick={onSaveEvent}>
      <AddCirculeIcon />
    </IconButton>
  ) : (
    <IconButton edge="end" aria-label="comments" onClick={onEditEvent}>
      <EditIcon />
    </IconButton>
  );
  return (
    <>
      {editionView}
      <IconButton
        edge="end"
        aria-label="comments"
        disabled={isEditionStatus}
        onClick={onDeleteEvent}
      >
        <DeleteIcon />
      </IconButton>
    </>
  );
}
