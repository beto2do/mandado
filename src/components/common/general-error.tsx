import { ErrorDetail } from '@/models'

export function GeneralError({error, reset}: ErrorDetail) {

    return (
        <div>
        <h2>Something went wrong!</h2>
        <button
          onClick={
            () => reset()
          }
        >
          Try again
        </button>
      </div>
    );
}