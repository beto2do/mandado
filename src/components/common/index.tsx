export * from "./element-icons";
export * from "./add-element";
export * from "./dashboard-tile";
export * from "./general-error";
export * from "./global-nav-bar";
export * from "./modal";
export * from "./skeleton-list";
export * from "./table-view";
export * from "./custom-modal";
export * from "./global-snackbar";
export * from "./logo-app";
export * from "./nav-menu";
export * from "./hamburger-menu";
export * from "./dashboard";
