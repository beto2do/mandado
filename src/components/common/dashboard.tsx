import { Tile } from "@/models";
import { DashboardTile } from "@/components/common";

export function Dashboard({ tiles }:{ tiles: Tile[] }) {

    const markupTiles = tiles.map((tile: Tile) => {
        return (
            <DashboardTile key={tile.title} tile={tile}></DashboardTile>
        );
      });

      return (
      <>
        {markupTiles}
        </>         
    )
}