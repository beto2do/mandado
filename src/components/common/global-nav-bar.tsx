"use client";

import { useState } from "react";
import AppBar from "@mui/material/AppBar";
import Toolbar from "@mui/material/Toolbar";
import Container from "@mui/material/Container";
import Link from "next/link";
import { useSelectedLayoutSegment } from "next/navigation";
import { LogoApp, NavMenu, HamburgerMenu } from "@/components/common";

const pages = [
  {
    label: "Shopping",
    path: "/shop",
  },
  {
    label: "Meals",
    path: "/meal",
  },
  {
    label: "Pantry",
    path: "/pantry",
  },
  {
    label: "Products",
    path: "/product",
  },
];

export function GlobalNavBar() {
  const segment = useSelectedLayoutSegment();
  const [anchorElNav, setAnchorElNav] = useState<null | HTMLElement>(null);

  const handleOpenNavMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElNav(event.currentTarget);
  };

  const handleCloseNavMenu = () => {
    setAnchorElNav(null);
  };

  return (
    <>
      <AppBar position="static">
        <Container maxWidth="xl">
          <Toolbar disableGutters>
            <LogoApp />
            <HamburgerMenu
              pages={pages}
              anchorElNav={anchorElNav}
              onClick={handleOpenNavMenu}
              onClose={handleCloseNavMenu}
            />
            <LogoApp mobile={true} />
            <NavMenu pages={pages} onClick={handleCloseNavMenu} />
            {segment !== "login" ? <Link href="/login">Login</Link> : null}
          </Toolbar>
        </Container>
      </AppBar>
    </>
  );
}
