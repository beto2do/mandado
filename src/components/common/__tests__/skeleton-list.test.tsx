import renderer from "react-test-renderer";
import { SkeletonList } from "@/components/common";

it("renders correctly", () => {
  const tree = renderer.create(<SkeletonList />).toJSON();
  expect(tree).toMatchSnapshot();
});
