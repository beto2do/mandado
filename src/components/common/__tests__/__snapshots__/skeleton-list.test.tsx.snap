// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <div
    className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      style={
        {
          "height": undefined,
          "width": undefined,
        }
      }
    />
  </div>,
  <div
    className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      style={
        {
          "height": undefined,
          "width": undefined,
        }
      }
    />
  </div>,
  <div
    className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      style={
        {
          "height": undefined,
          "width": undefined,
        }
      }
    />
  </div>,
  <div
    className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
      style={
        {
          "height": undefined,
          "width": undefined,
        }
      }
    />
  </div>,
]
`;
